pyproject.toml

```diff
@@ -14,6 +14,8 @@ dynamic = ["version"]
 license = {text = "BSD-3-Clause"}
 
 dependencies = [
+  "tables",
+  "blosc2",
```

Comment:
☝️  After merging `main` into this branch in https://github.com/SainsburyWellcomeCentre/WAZP/pull/36/commits/4ff5406dd91179d410f97b0a70c9d8ca66cbdb7a (then I fixed a trivial error), it seems the smoke test _actually flagged up_ a missing dependency (👍).

It's an [error](https://github.com/SainsburyWellcomeCentre/WAZP/actions/runs/4306662767/jobs/7510984926#step:3:763) about [missing optional pytables](https://pandas.pydata.org/docs/getting_started/install.html#other-data-sources). So this also fixes that.

I think this makes sense as we've added some HDF5 input since I made this branch.

Reply:
The problem about `pytables` is that `pip install` doesn't work on many machines (Adam mentioned this might be due to missing hdf5 library, which pip cannot install).
That's why I have recommended creating a `conda` environment with `pytables` in [install instructions](https://sainsburywellcomecentre.github.io/WAZP/) (`conda install` works).

Does "blosc2" solve the above issue? If yes, we can simply get rid of the whole `conda` thing

Reply:
No. blocsc2 is some compression for hdf5. It won't solve any (py)tables installation woes.

Trouble is, for the tests we'll need `tables` (it's fine for the tests because hdf5 is on the runners).

Options:
1. Leave it like this. (I think I prefer this option because it's "pytables' fault" if it won't install.
      - Perhaps add a "you might have errors from pytables, here's where to go for help"-type sentence to the docs.
3. Move `tables` to a dev dependency (which it really isn't, so I don't like this). But could then add an explicit `conda` installation step of tables to the instructions.
4. @dstansby showed me [this](https://github.com/tox-dev/tox-conda) which might help.

Reply:
That's annoying.
 Tagging @adamltyson because we have discussed this before, and it's also relevant to other NIU projects.
**Question:** if we add `tables` as a regular (pip) dependency, will `pip install wazp` fail when users run it? Or will it just install wazp without `pytables` and let them know that the latter is missing?

Reply:
Sort of dodging the question, but if tables is going to be necessary for proper functioning of the app, then I would target conda-forge as the main release mechanism.

Reply:
My (univited but I've been tagged so sort of invited) two cents here is it would be best specify `pandas[hdf5]` if you're only loading HDF5 via pandas, and not explicitly have `tables` as a dependency. This way if pandas change the library they're using to load HDF5 files you don't have to change anything.

re. installing a hdf5 library:
- For testing, I've used `tox-conda` before and would recommend.
- For users I agree that targeting `conda-forge` as the main release channel is a good way to go. This way you can specify a hdf5 dependency in the conda-forge recipe.

And finally, FWIW, if you try and install `tables` on a platform that doesn't have wheels and tries to build from source, and don't have a HDF5 library you get:
```python
Collecting tables
  Using cached tables-3.8.0.tar.gz (8.0 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [11 lines of output]
      cpuinfo failed, assuming no CPU features: 'flags'
      * Using Python 3.11.0 | packaged by conda-forge | (main, Jan 14 2023, 12:26:40) [Clang 14.0.6 ]
      * Found cython 0.29.33
      * USE_PKGCONFIG: True
      * Found conda env: ``/Users/dstansby/mambaforge/envs/pytables``
      ld: library not found for -lhdf5
      clang: error: linker command failed with exit code 1 (use -v to see invocation)
      .. ERROR:: Could not find a local HDF5 installation.
         You may need to explicitly state where your local HDF5 headers and
         library can be found by setting the ``HDF5_DIR`` environment
         variable or by using the ``--hdf5`` command-line option.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

Looking at the [PyPI files for `tables`](https://pypi.org/project/tables/#files) this just looks like it's because there's no wheel for macOS arm, so here I'd say asking for one of those upstream in the `tables` repo might be a good thing to do.

Reply:
> it would be best specify pandas[hdf5] if you're only loading HDF5 via pandas
> [...] This way if pandas change the library they're using to load HDF5 files you don't have to change anything.

That would be nice but I don't think we can have nice things...
```
(wazp-env) ➤ grep dependencies\ = pyproject.toml
dependencies = ["pandas[hdf5]", "dash", "dash-bootstrap-components"]
(wazp-env) ➤ tox -e py310 -vv 
```
eventually you get a:
```
WARNING: pandas 1.5.3 does not provide the extra 'hdf5'
```

About the pyTables wheel, seems like they did make it: https://github.com/PyTables/PyTables/issues/887 and https://github.com/PyTables/PyTables/pull/944.

Reply:
I have a silicon Mac, so I created a fresh `conda` environment, ran `pip install  .` on this branch, and got the error that David posted. I tried the same on an ubuntu machine and it worked (presumably because there is a system-level install of hdf5 ?).
Anyhow, at least some users will get the error when running `pip install wazp`.

So how about this action plan:
* We release on pip as is, but users have to follow the current installation instructions:
  ```sh
  conda create -n wazp-env -c conda-forge python=3 pytables
  conda activate wazp-env
  pip install wazp
  ```
  (alternatively we could create a conda `environment.yml` file)
* In parallel we start the process of releasing on `conda-forge`
* Re testing, Sam decides how to handle it, because I don't understand it well enough to offer an opinion

Reply:
The test job should be as-close-as-possible environment to the users. So if we install `tables` via `conda-force`, then the GH workflow ideally does that.

I'd suggest merging this PR as-is. I claim that installing via `conda` as per the instructions, and _then_ reinstalling as a dependency via `pip` works. Even though it's double-work.

We can fix `pyproject.toml` accordingly later if we fix the GH workflow or release some other way or...